chains/evm/contracts/onRamp/ExecutorOnRamp.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IExecutorOnRamp} from "../interfaces/IExecutorOnRamp.sol";
+
+import {Client} from "../libraries/Client.sol";
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+
+import {EnumerableSet} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v5.0.2/contracts/utils/structs/EnumerableSet.sol";
+
+/// @notice The ExecutorOnRamp configures the supported destination chains and CCV limits for an executor.
+contract ExecutorOnRamp is Ownable2StepMsgSender, IExecutorOnRamp {
+  using EnumerableSet for EnumerableSet.AddressSet;
+  using EnumerableSet for EnumerableSet.UintSet;
+
+  error ExceedsMaxCCVs(uint256 provided, uint256 max);
+  error InvalidCCV(address ccv);
+  error InvalidDestChain(uint64 destChainSelector);
+  error InvalidMaxPossibleCCVsPerMsg(uint256 maxPossibleCCVsPerMsg);
+
+  event AllowlistUpdated(bool enabled);
+  event CCVAdded(address indexed ccv);
+  event CCVRemoved(address indexed ccv);
+  event ConfigSet(DynamicConfig dynamicConfig);
+  event DestChainAdded(uint64 indexed destChainSelector);
+  event DestChainRemoved(uint64 indexed destChainSelector);
+
+  /// @dev Struct that defines the dynamic configuration.
+  // solhint-disable-next-line gas-struct-packing
+  struct DynamicConfig {
+    // Max(required ccvs + optional ccvs).
+    // Limits the number of ccvs that the executor needs to search for results from.
+    uint8 maxCCVsPerMsg;
+  }
+
+  /// @notice Whether or not the CCV allowlist is enabled.
+  bool public s_allowlistEnabled;
```

Comment:
Can we rename this to s_ccvAllowlistEnabled? Allowlists normally mean the msg sender